GitUI/UserControls/RevisionGrid/Graph/GraphModel.cs

```diff
@@ -746,7 +757,15 @@ public void Prune()
             // Remove all nodes that don't have a value associated with them.
             foreach (var node in nodesToRemove)
             {
-                _nodeByObjectId.Remove(node.ObjectId);
+                if (!_nodeByObjectId.TryRemove(node.ObjectId, out Node removedValue))
+                {
+                    if (Debugger.IsAttached)
+                    {
+                        Debugger.Break();
+                    }
+
+                    _nodeByObjectId.TryRemove(node.ObjectId, out removedValue);
+                }
```

Comment:
I'm having troubles understanding these `if`s.
Could you please clarify why `TryAdd` and `TryRemove` are duplicated?

Reply:
Removed